### PR TITLE
DNSSec on Managed Zones

### PR DESCRIPTION
**Pull request by rambleraptor**
Adding DNSSec support to Managed Zones in response to https://github.com/ansible/ansible/issues/46050

-----------------------------------------------------------------
# [all]

### CODE REVIEW

**Comment by rambleraptor** [comment]
@SirGitsalot - normal code review
@danawillow - TF stuff (some of the PRs are because of drift, but there are valid DNS changes in the PRs)

**Comment by danawillow** [comment]
Looks like a dupe of https://github.com/GoogleCloudPlatform/magic-modules/pull/1733, should one be closed in favor of the other?

**Comment by rambleraptor** [comment]
+adding chrisst@ who was handling the other DNSSEC PR.

File: third_party/terraform/tests/resource_dns_managed_zone_test.go.erb [hunk 1]
```diff
@@ -128,6 +128,9 @@ resource "google_dns_managed_zone" "private" {
     networks {
       network_url = "${google_compute_network.%s.self_link}"
     }
+    dnssec_config {
+      state = "off"
```
Comment (chrisst) [inline third_party/terraform/tests/resource_dns_managed_zone_test.go.erb]: Is there a way to test this in a 'on' state? I'm not super familiar with the dnssec but this doesn't look like it tests that DNS can be configured by Terraform.

Comment (rambleraptor) [reply third_party/terraform/tests/resource_dns_managed_zone_test.go.erb]: Yeah, that's a good point. Just pushed up a (passing!) version of the test that has dnssec on.

File: products/dns/api.yaml [hunk 2]
```diff
@@ -55,6 +55,64 @@ objects:
...
+        name: 'dnssecConfig'
+        description: DNSSEC configuration
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'kind'
+            description: Identifies what kind of resource this is
+            default_value: 'dns#managedZoneDnsSecConfig'
+          - !ruby/object:Api::Type::Enum
+            name: 'nonExistence'
+            description: >
```
[the remark this answers is not in the document]
Comment (rambleraptor) [reply products/dns/api.yaml]: Yes! `|` keeps newlines in place while `>` removes them. Using `|` means we have a whole bunch of weird, stray newline characters.

In almost all situations, I think we should burn `|` with fire.

**Comment by rambleraptor** [comment]
@chrisst can you merge the TF PRs?

**Pull request merged**